src/polyswarmd/rpc.py

```diff
@@ -0,0 +1,237 @@
+                    reveal['data']['metadata'] = substitute_ipfs_metadata(reveal['data'].get('metadata', ''))
+
+                    self.broadcast(reveal)
+
+                for event in self.vote_filter.get_new_entries():
+                    vote = {
+                        'event': 'vote',
+                        'data': new_vote_event_to_dict(event.args),
+                        'block_number': event.blockNumber,
+                        'txhash': event.transactionHash.hex(),
+                    }
+                    self.broadcast(vote)
+
+                for event in self.quorum_filiter.get_new_entries():
+                    quorum = {
+                        'event': 'quorum',
+                        'data': new_quorum_event_to_dict(event.args),
+                        'block_number': event.blockNumber,
+                        'txhash': event.transactionHash.hex(),
+                    }
+                    self.broadcast(quorum)
+
+                for event in self.settled_filter.get_new_entries():
+                    settled_bounty = {
+                        'event': 'settled_bounty',
+                        'data': settled_bounty_event_to_dict(event.args),
+                        'block_number': event.blockNumber,
+                        'txhash': event.transactionHash.hex(),
+                    }
+                    self.broadcast(settled_bounty)
+
+                for event in self.deprecated_filter.get_new_entries():
+                    deprecated = {
+                        'event': 'deprecated',
+                        'data': deprecated_event_to_dict(event.args),
+                        'block_number': event.blockNumber,
+                        'txhash': event.transactionHash.hex()
+                    }
+                    self.broadcast(deprecated)
+
+                if self.init_filter is not None:
+                    for event in self.init_filter.get_new_entries():
+                        initialized_channel = {
+                            'event': 'initialized_channel',
+                            'data': new_init_channel_event_to_dict(event.args),
+                            'block_number': event.blockNumber,
+                            'txhash': event.transactionHash.hex(),
+                        }
+                        self.broadcast(initialized_channel)
+                for _ in self.block_filter.get_new_entries():
+                    block = {
+                        'event': 'block',
+                        'data': {
+                            'number': self.chain.w3.eth.blockNumber,
+                        },
+                    }
+                    self.broadcast(block)
+            except ConnectionError:
+                logger.exception('ConnectionError in filters (is geth down?)')
+                continue
+            except Exception:
+                logger.exception('Exception in filter checks, resetting filters')
+                self.flush_filters()
```

Comment:
This is incorrect here. It definitely needs more robust handling. 
I have seen a few errors, unfortunately I did not record them. 
An inprogress list is below
```
ValueError: filter not found lost a filter after a long period with no websockets. 
```

Reply:
It does recover correctly from runtime errors, like the error mentioned in the main blob, where substitute ipfs metadata is being called from the wrong greenlet.